api/v1alpha1/mutation_types.go

```diff
@@ -0,0 +1,96 @@
+package v1alpha1
+
+import (
+	"time"
+
+	"github.com/fluxcd/pkg/apis/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// MutationSpec defines a common spec between Localization and Configuration.
+type MutationSpec struct {
```

Comment:
mutation is a good term 👍🏻

Reply:
Should we add componentversion as part of the `MutationSpec`? Currently we are taking it from the `ConfigRef`  which is not right, that should only be for specify the component from which to fetch the config.

I think` spec.source` should support either kind=Source or kind=ComponentVersion; that is to say if the source is not a snapshot CR then it should be a componentversion CR.

Reply:
I agree. But let's do that in a separate PR. I feel like that could get bigger. :D